Compatibility break in `couchdb_session_auth` with moving core from 3.3.6 to 3.5.0

There seems to be a compatibility break in `couchdb_session_auth` module with moving core from v3.3.6 to v3.5.0. A test failure is consistent for all python versions and in both Travis and Jenkins:

```
[2021-01-27T13:57:19.250Z] test/unit/test_couchdb_session_auth.py ....RRRF...                       [100%]
[2021-01-27T13:57:19.250Z] 
[2021-01-27T13:57:19.250Z] =================================== FAILURES ===================================
[2021-01-27T13:57:19.250Z] ___________ TestCouchDbSessionAuth.test_disable_ssl_verification_on ____________
[2021-01-27T13:57:19.251Z] 
[2021-01-27T13:57:19.251Z] self = <test_couchdb_session_auth.TestCouchDbSessionAuth testMethod=test_disable_ssl_verification_on>
[2021-01-27T13:57:19.251Z] args = (<MagicMock name='request' id='140154731883880'>,)
[2021-01-27T13:57:19.251Z] 
[2021-01-27T13:57:19.251Z]     @responses.activate
[2021-01-27T13:57:19.251Z]     @mock.patch('requests.request', wraps=requests_request_spy)
[2021-01-27T13:57:19.251Z]     def test_disable_ssl_verification_on(self, *args):
[2021-01-27T13:57:19.251Z]         global request_kwargs
[2021-01-27T13:57:19.251Z]         self.client.set_disable_ssl_verification(True)
[2021-01-27T13:57:19.251Z]         self.client.get_session_information()
[2021-01-27T13:57:19.251Z] >       assert request_kwargs['verify'] is False
[2021-01-27T13:57:19.251Z] E       KeyError: 'verify'
```

Ref: https://github.com/IBM/cloudant-python-sdk/pull/66

**To Reproduce**
1. Change `requirements.txt` to `ibm_cloud_sdk_core==3.5.0`
2. Run `python3 -m tox -e py36`

**Expected behavior**
Tests should pass

--- a/requirements.txt
+++ b/requirements.txt
@@ -2,4 +2,4 @@ jproperties
 requests>=2.0,<3.0
 python_dateutil>=2.5.3
 websocket-client==0.57.0
-ibm_cloud_sdk_core==3.3.6
+ibm_cloud_sdk_core==3.5.0

--- a/test/unit/test_couchdb_session_auth.py
+++ b/test/unit/test_couchdb_session_auth.py
@@ -31,11 +31,15 @@ request_args = None
 request_kwargs = None
 
 
-def requests_request_spy(*args, **kwargs):
-    global request_args, request_kwargs
-    request_args = args
-    request_kwargs = kwargs
-    return request(*args, **kwargs)
+class MockSession(requests.sessions.Session):
+    def __init__(self):
+        super().__init__()
+        self.last_verify = None
+
+    def request(self, *args, **kwargs):
+        if 'verify' in kwargs:
+            self.last_verify = kwargs['verify']
+        return super().request(*args, **kwargs)
 
 
 class TestCouchDbSessionAuth(unittest.TestCase):
@@ -80,20 +84,28 @@ class TestCouchDbSessionAuth(unittest.TestCase):
         self.assertEqual(responses.calls[1].request.headers["yes"], "works")
 
     @responses.activate
-    @mock.patch('requests.request', wraps=requests_request_spy)
-    def test_disable_ssl_verification_on(self, *args):
-        global request_kwargs
-        self.client.set_disable_ssl_verification(True)
-        self.client.get_session_information()
-        assert request_kwargs['verify'] is False
+    def test_disable_ssl_verification_on(self):
+        original_http_client = self.client.get_http_client()
+        mock_session = MockSession()
+        try:
+            self.client.set_http_client(mock_session)
+            self.client.set_disable_ssl_verification(True)
+            self.client.get_session_information()
+            assert mock_session.last_verify is False
+        finally:
+            self.client.set_http_client(original_http_client)
 
     @responses.activate
-    @mock.patch('requests.request', wraps=requests_request_spy)
-    def test_disable_ssl_verification_off(self, *args):
-        global request_kwargs
-        self.client.set_disable_ssl_verification(False)
-        self.client.get_session_information()
-        self.assertNotIn('verify', request_kwargs)
+    def test_disable_ssl_verification_off(self):
+        original_http_client = self.client.get_http_client()
+        mock_session = MockSession()
+        try:
+            self.client.set_http_client(mock_session)
+            self.client.set_disable_ssl_verification(False)
+            self.client.get_session_information()
+            assert mock_session.last_verify is None
+        finally:
+            self.client.set_http_client(original_http_client)
 
     @responses.activate
     def test_cookie_refresh(self):

--- a/test/unit/test_couchdb_session_auth.py
+++ b/test/unit/test_couchdb_session_auth.py
@@ -17,10 +17,8 @@
 import datetime
 import json
 import unittest
-from unittest import mock
 
 import requests
-import responses
 import time
 
 from ibmcloudant import CouchDbSessionAuthenticator

--- a/test/unit/test_couchdb_session_auth.py
+++ b/test/unit/test_couchdb_session_auth.py
@@ -19,6 +19,7 @@ import json
 import unittest
 
 import requests
+import responses
 import time
 
 from ibmcloudant import CouchDbSessionAuthenticator

